### DISCUSSION TITLE
Library Import Error When Running 'safety-gymnasium/examples/multi_goal.py'

### DISCUSSION THREAD

**Discussion by JingjieNi** [PKU-Alignment/safety-gymnasium#127] [Q&A]
Hello, I tried to run 'safety-gymnasium/examples/multi_goal.py' and encountered an error. After checking, I found that safety_gymnasium.tasks has been installed under the path 'C:\Users\nijin\safety-gymnasium\safety_gymnasium'. 
How can I debug this issue?

Traceback (most recent call last):
File `~\anaconda3\envs\s\afyGYM\lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
exec(code, globals, locals)
File `c:\users\nijin\desktop\desktop\safegym\cus_env_test\multienvtest.py:44 in run_random
env = safety_gymnasium.make(env_name, render_mode='human')
File `c:\users\nijin\desktop\desktop\safegym\cus_env_test\multienv.py:15 in run_random
env = safety_gymnasium.make(env_name, render_mode='human')
File `~\safety-gymnasium\safety_gymnasium\utils\registration.py:126 in make
env_creator = load_env_creator(env_spec.entry_point)
File `~\anaconda3\envs\afyGYM\lib\site-packages\gymnasium\envs\registration.py:553 in load_env_creator
mod = importlib.import_module(mod_name)
File `~\anaconda3\envs\afyGYM\lib\importlib__init__.py:127 in import_module
return _bootstrap._gcd_import(name[level:], package, level)
File `<frozen importlib._bootstrap>:1014 in _gcd_import
File `<frozen importlib._bootstrap>:991 in _find_and_load
File `<frozen importlib._bootstrap>:971 in _find_and_load_unlocked
File `<frozen importlib._bootstrap>:914 in _find_spec
File `<frozen importlib._bootstrap_external>:1407 in find_spec
File `<frozen importlib._bootstrap_external>:1373 in _get_spec
File `<frozen importlib._bootstrap_external>:1239 in iter
File `<frozen importlib._bootstrap_external>:1227 in _recalculate
File `<frozen importlib._bootstrap_external>:1223 in _get_parent_path
KeyError: 'safety_gymnasium.tasks'

**Comment by JingjieNi** [comment]
Hi, I've noticed that the script throws an error on the second run, but the issue is resolved by opening a new console.

**Reply by muchvo** [reply]
It looks like you’re running `safety-gymnasium` on Windows, and based on the error, it may be related to the environment itself. If possible, we recommend running it within WSL2 for Ubuntu 20.04 on Windows, which has been thoroughly tested on our end. If the issue persists, please provide additional details to help us reproduce and troubleshoot this problem.

**Reply by JingjieNi** [reply]
Thank you for your suggestions. I am indeed running the code on a Windows system. I've found that every time I open a new console, I can successfully run the single-agent and multi-agent environments in SafeGYM. At the same time, these environments can also be successfully connected with the Omnisafe algorithm library you provided and the OpenRL algorithm library (https://github.com/OpenRL-Lab/openrl). Your code has strong compatibility.

**Reply by muchvo** [reply]
I'm glad this repo has been helpful to you! Feel free to ask any questions.
